cli_tools/gce_vm_image_import/importer/importer.go

```diff
@@ -137,14 +137,14 @@ func (i *importer) runProcess(ctx context.Context) error {
 func (i *importer) runStep(ctx context.Context, step func() error, cancel func(string) bool, getTraceLogs func() []string) (err error) {
 	e := make(chan error)
 	var wg sync.WaitGroup
-	wg.Add(1)
 	go func() {
 		//this select checks if context expired prior to runStep being called
 		//if not, step is run
 		select {
 		case <-ctx.Done():
 			e <- i.getCtxError(ctx)
 		default:
+			wg.Add(1)
 			stepErr := step()
 			wg.Done()
```

Comment:
in order to fix the potential deadlock better, I suggest to use "defer wg.done()" in case step() triggered a panic. There are do some panic code added.

Reply:
if `wg.Done()` was called from defer, it would be invoked even if the first case happened (ctx.Done()). That would result in a panic of it's own due to negative wg counter.

Also, deferred is called even if panic occurs. E.g. this code:
'''
func main() {
	defer fmt.Println("deferred")
	panic("panicking")
}
'''

Will print:
'''
deferred
panic: panicking
'''

Reply:
correct. Let me give a LGTM then you decide modify or not

Reply:
Done